src/impl/results_notifier.hpp

```diff
@@ -61,6 +64,15 @@ class ResultsNotifier : public CollectionNotifier {
     // The changeset calculated during run() and delivered in do_prepare_handover()
     CollectionChangeBuilder m_changes;
     TransactionChangeInfo* m_info = nullptr;
+    partial_sync::SubscriptionState m_previous_partial_sync_state = partial_sync::SubscriptionState::Undefined;
+
+    // Key used for this partial sync subscription
+    std::string m_partial_sync_name;
```

Comment:
This design does not work for named subscriptions as there is not a 1:! correspondence between ResultsNotifier instances and subscription names.

Reply:
Not sure I understand you, but are you saying that a given instance of a `results_notifier.hpp` is used by multiple `Results`? Which mean that these states should be moved to `Results` instead and then read or sent as they are needed?

Reply:
Results should support calling subscribe() on a single instance more than one time on a single instance with different names.

Reply:
That doesn't seem compatible with getting changes reported back through the notification system? As you would then get changes back from multiple subscriptions, which would be really confusing?

Reply:
Why would subscribing to the same query multiple times (whether through a single Results or multiple) have any impact on what changes get reported?

Reply:
Any notifications related to partial sync state changes would potentially come in multiple times as the server may initially see each of the subscriptions at different times. This seems unavoidable if we're trying to force partial sync state changes through the Results notification mechanism.

Reply:
I guess you got a point since any errors at this point would only be along the line of "Server rejected the query" which all subscribers would need to know about.